client/lib/posts/post-list-store.js

```diff
@@ -1,371 +1,385 @@
-	_activeList.nextPageHandle = data.meta && data.meta.next_page;
+		if ( error ) {
+			debug( 'Error fetching PostsList from api:', error );
+			_activeList.errors.push( error );
+			return;
+		}
 
-	if ( ! _activeList.nextPageHandle ) {
-		_activeList.isLastPage = true;
-	}
+		if ( ! found ) {
+			_activeList.isLastPage = true;
+			return;
+		}
+
+		// if we got a next page handle, cache it for the next page
+		_activeList.nextPageHandle = data.meta && data.meta.next_page;
 
-	posts = data.posts;
+		if ( ! _activeList.nextPageHandle ) {
+			_activeList.isLastPage = true;
+		}
 
-	postIds = posts.map( function( post ) {
-		return post.global_ID;
-	} );
+		posts = data.posts;
+
+		postIds = posts.map( function( post ) {
+			return post.global_ID;
+		} );
 
-	if ( postIds.length ) {
-		// did we actually find any new posts?
-		postIds = difference( postIds, _activeList.postIds );
 		if ( postIds.length ) {
-			_activeList.postIds = _activeList.postIds.concat( postIds );
-			_activeList.page++;
+			// did we actually find any new posts?
+			postIds = difference( postIds, _activeList.postIds );
+			if ( postIds.length ) {
+				_activeList.postIds = _activeList.postIds.concat( postIds );
+				_activeList.page++;
+			}
 		}
 	}
 
-	PostListStore.emit( 'change' );
-}
+	// Merge updated posts
+	function receiveUpdates( listId, error, data ) {
+		let posts;
+		let postIds;
+		let newPostIds;
 
-/**
- * Merge updated posts
- **/
-function receiveUpdates( id, error, data ) {
-	var posts, postIds, newPostIds;
-
-	if ( error ) {
-		debug( 'An error occurred while fetching updated posts %o', error );
-		return;
-	}
+		if ( error ) {
+			debug( 'An error occurred while fetching updated posts %o', error );
+			return;
+		}
 
-	if ( id !== _activeList.id ) {
-		return;
-	}
+		if ( listId !== _activeList.id ) {
+			return;
+		}
 
-	if ( ! data.posts.length ) {
-		return;
-	}
+		if ( ! data.posts.length ) {
+			return;
+		}
 
-	posts = data.posts;
+		posts = data.posts;
 
-	debug( 'Fetched updated posts:', posts );
+		debug( 'Fetched updated posts:', posts );
 
-	postIds = posts.map( function( post ) {
-		return post.global_ID;
-	} );
+		postIds = posts.map( function( post ) {
+			return post.global_ID;
+		} );
 
-	newPostIds = difference( postIds, _activeList.postIds );
+		newPostIds = difference( postIds, _activeList.postIds );
 
-	if ( newPostIds.length ) {
-		_activeList.postIds = _activeList.postIds.concat( newPostIds );
-		sort( _activeList.postIds );
+		if ( newPostIds.length ) {
+			_activeList.postIds = _activeList.postIds.concat( newPostIds );
+			sort( _activeList.postIds );
+		}
 	}
 
-	PostListStore.emit( 'change' );
-}
+	return new class extends EventEmitter {
+		constructor() {
+			super();
+			this.id = id;
+			this.dispatchToken = Dispatcher.register( this.handlePayload.bind( this ) );
+		}
 
-PostListStore = {
+		get() {
+			return _activeList;
+		}
 
-	get: function() {
-		return _activeList;
-	},
+		getID() {
```

Comment:
Tempting to use this opportunity to fix the [naming convention errors](https://github.com/Automattic/wp-calypso/blob/master/docs/coding-guidelines/javascript.md#naming-conventions) here, though I'm inclined to avoid it in this already-large pull request.